tests/integration/targets/docker_container/tasks/tests/options.yml

```
@@ -2534,6 +2560,10 @@ avoid such warnings, please quote the value.' in log_options_2.warnings"
     # on a proper VM), memory_swap cannot be set and will be -1 afterwards.
     - memory_swap_2 is not changed or memory_swap_2.container.HostConfig.MemorySwap == -1
     - memory_swap_3 is changed
+    # Unlimited memory_swap (using 'unlimited') should be allowed
+    - memory_swap_4 is changed and memory_swap_4.container.HostConfig.MemorySwap == -1
```

```suggestion
    # (If the value was already -1 because of the above reasons, it won't change)
    - (memory_swap_4 is changed or memory_swap_3.container.HostConfig.MemorySwap == -1) and memory_swap_4.container.HostConfig.MemorySwap == -1
```
This is not surprising (from the above comments). The tests passed on the VMs (the RHEL CI nodes are VMs, the others are docker containers), so they work fine in principle :)